File: leetcode/problems/js/find-the-highest-altitude.js
// by @codeAbinash
// Time : O(n)
// Space : O(1)

function largestAltitude(gain) {
  let altitude = 0
  let max = 0
  for (let g of gain) {
    altitude += g
    max = Math.max(max, altitude)
  }
  return max
}

File: leetcode/problems/rs/find-the-highest-altitude.rs
// by @codeAbinash
// Time : O(n)
// Space : O(1)

impl Solution {
    pub fn largest_altitude(gain: Vec<i32>) -> i32 {
        let mut altitude = 0;
        let mut max_altitude = 0;
        for i in gain {
            altitude += i;
            max_altitude = max_altitude.max(altitude);
        }
        max_altitude
    }
}
